include/kes_dashboard_live_tiles_client/TileCache.h

```diff
@@ -0,0 +1,63 @@
+/**
+ * \file TileCache.h
+ *
+ * \copyright Copyright (C) 2019 Kano Computing Ltd.
+ *            License: http://www.gnu.org/licenses/gpl-2.0.txt GNU GPL v2
+ *
+ * \brief     TODO
+ */
+
+
+#ifndef INCLUDE_KES_DASHBOARD_LIVE_TILES_CLIENT_TILECACHE_H_
+#define INCLUDE_KES_DASHBOARD_LIVE_TILES_CLIENT_TILECACHE_H_
+
+
+#include <list>
+#include <memory>
+#include <string>
+
+#include "kes_dashboard_live_tiles_client/ITile.h"
+#include "kes_dashboard_live_tiles_client/ITileCache.h"
+#include "kes_dashboard_live_tiles_client/ITileFactory.h"
+#include "kes_dashboard_live_tiles_client/TileFactory.h"
+
+
+namespace KESDLTC {
+namespace internal {
+
+class TileCache : public ITileCache {
+ public:  // Constructors & destructors.
+    explicit TileCache(
+        const std::string& cacheDir,
+        const std::shared_ptr<ITileFactory> tileFactory =
+            std::make_shared<TileFactory>());
+    ~TileCache();
+
+ public:  // ITileLoader Methods.
+    std::list<std::shared_ptr<ITile>> getTiles() override;
+
+ public:  // ITileCache Methods.
+    void update(const std::list<std::shared_ptr<ITile>>& tiles) override;
```

Comment:
Does passing by reference actually make a difference? It is already basically pointer to pointers so not really much, if any, saved here.

Reply:
I left that as is back from when it was `const std::list<Tile>& tiles`. Which one do you think would be better?

```
const std::list<std::shared_ptr<ITile>>& tiles
```
or
```
std::list<std::shared_ptr<ITile>> tiles
```